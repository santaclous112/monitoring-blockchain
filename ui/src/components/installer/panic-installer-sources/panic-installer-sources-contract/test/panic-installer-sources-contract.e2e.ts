import {newSpecPage} from '@stencil/core/testing';
import {
  PanicInstallerSourcesContract
} from "../panic-installer-sources-contract";
import {
  PanicInstallerSourcesContractForm
} from "../panic-installer-sources-contract-form";
import {
    PanicInstallerSourcesContractWeiWatcher
} from "../panic-installer-sources-contract-wei-watcher";

describe('panic-installer-sources-contract', () => {
  it('renders', async () => {
    const page = await newSpecPage(
        {
          components: [PanicInstallerSourcesContract],
          html: '<panic-installer-sources-contract></panic-installer-sources-contract>',
        }
    );

    const element = page.body.querySelector('panic-installer-sources-contract');
    expect(element).toMatchSnapshot();
  });
});

describe('panic-installer-sources-contract-wei-watcher', () => {
  it('renders', async () => {
    const page = await newSpecPage(
        {
            components: [PanicInstallerSourcesContractWeiWatcher],
            html: '<panic-installer-sources-contract-wei-watcher></panic-installer-sources-contract-wei-watcher>',
        }
    );

    const element = page.body.querySelector('panic-installer-sources-contract-wei-watcher');
    expect(element).toMatchSnapshot();
  });
});

describe('panic-installer-sources-contract-form', () => {
  it('renders', async () => {
    const page = await newSpecPage(
        {
          components: [PanicInstallerSourcesContractForm],
          html: '<panic-installer-sources-contract-form></panic-installer-sources-contract-form>',
        }
    );

    const element = page.body.querySelector('panic-installer-sources-contract-form');
    expect(element).toMatchSnapshot();
  });
});
