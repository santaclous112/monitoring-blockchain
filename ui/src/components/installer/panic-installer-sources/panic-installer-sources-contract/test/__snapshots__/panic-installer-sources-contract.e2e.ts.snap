// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panic-installer-sources-contract renders 1`] = `
<panic-installer-sources-contract>
  <svc-surface class="panic-installer-sources__container">
    <svc-label class="panic-installer-sources__title">
      Contracts Monitoring Setup
    </svc-label>
    <div class="panic-installer-sources__more-info">
      <p>
        Here you will configure the contracts you want monitored and triggering alerts
      </p>
      <svc-buttons-container>
        <svc-button color="secondary" iconname="information-circle" iconposition="icon-only"></svc-button>
      </svc-buttons-container>
    </div>
    <div class="panic-installer-sources-contract__container-form"></div>
    <div class="panic-installer-sources-contract__container-wei">
      <panic-installer-sources-contract-wei-watcher></panic-installer-sources-contract-wei-watcher>
    </div>
  </svc-surface>
</panic-installer-sources-contract>
`;

exports[`panic-installer-sources-contract-form renders 1`] = `
<panic-installer-sources-contract-form>
  <form>
    <svc-input label="Custom Network Name" labelcolor="primary" lines="inset" name="name" placeholder="Enter custom network name..." required=""></svc-input>
    <svc-input label="Custom Network URL" labelcolor="primary" lines="inset" name="url" placeholder="Enter custom network URL..." required="" type="url"></svc-input>
    <div class="panic-installer-sources-contract__add-contract-buttons">
      <svc-button color="primary" iconname="checkmark" type="submit">
        Save
      </svc-button>
      <svc-button color="primary" iconname="close">
        Cancel
      </svc-button>
    </div>
  </form>
</panic-installer-sources-contract-form>
`;

exports[`panic-installer-sources-contract-wei-watcher renders 1`] = `
<panic-installer-sources-contract-wei-watcher>
  <div class="panic-installer-sources-contract__network-add-container">
    <svc-select header="Network Name" name="name" placeholder="Network Name" withborder=""></svc-select>
    <svc-buttons-container>
      <svc-button color="primary" iconname="add-circle" iconposition="icon-only"></svc-button>
    </svc-buttons-container>
  </div>
  <div>
    <svc-input disabled="" label="WeiWatchers URL" labelcolor="primary" lines="inset" name="url"></svc-input>
    <svc-toggle helptext="Whether to monitor price feed contract for this chain." htmlid="monitor_contracts" label="Enable Contract Monitoring" lines="none" name="monitor" position="end" value=""></svc-toggle>
  </div>
</panic-installer-sources-contract-wei-watcher>
`;
